modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Monad.kt

```diff
@@ -99,6 +99,9 @@ interface Monad<F> : Selective<F> {
     flatMap { it.fold({ a -> f.map { ff -> ff(a) } }, { b -> just(b) }) }
 
   override fun <A, B> Kind<F, Either<A, B>>.select(f: Kind<F, (A) -> B>): Kind<F, B> = selectM(f)
+
+  override fun <A, B> Kind<F, A>.lazyAp(ff: () -> Kind<F, (A) -> B>): Kind<F, B> =
```

Comment:
Because flatMap is actually lazy for all types that can short circuit it is a really good fit for a default and because of monad-applicative consistency laws this is also always the same behaviour